src/main/java/io/jenkins/pluginhealth/scoring/scores/DeprecatedPluginScoring.java

```diff
@@ -37,10 +39,15 @@ public class DeprecatedPluginScoring extends Scoring {
     private static final float COEFFICIENT = .8f;
     private static final String KEY = "deprecation";
 
+
+
     @Override
     protected ScoreResult doApply(Plugin plugin) {
         final ProbeResult deprecatedPluginProbeResult = plugin.getDetails().get(DeprecatedPluginProbe.KEY);
-        if (deprecatedPluginProbeResult == null || deprecatedPluginProbeResult.status().equals(ResultStatus.FAILURE)) {
+
+        final ProbeResult doesPluginExistInUpdateCenterMapResult = plugin.getDetails().get(UpdateCenterPluginPublicationProbe.KEY);
+
+        if (deprecatedPluginProbeResult == null || deprecatedPluginProbeResult.status().equals(ResultStatus.FAILURE) || doesPluginExistInUpdateCenterMapResult.status().equals(ResultStatus.FAILURE) ) {
```

Comment:
You need to make sure this new ProbeResult instance is not null.

```suggestion
        if (deprecatedPluginProbeResult == null || deprecatedPluginProbeResult.status().equals(ResultStatus.FAILURE) || doesPluginExistInUpdateCenterMapResult == null || doesPluginExistInUpdateCenterMapResult.status().equals(ResultStatus.FAILURE) ) {
```